0350-cmake-re-cli-reference.md

```diff
@@ -0,0 +1,131 @@
+---
+title: 👩🏼‍💻 cmake-re --help | Command Line Reference
+---
+
+**CMake RE :** CMake Remote Execution, transparent cmake wrapper with build isolation and caching capabilities.
+
+If you know to use `cmake`, using `cmake-re` is a drop-in replacement which runs cached and hermetic builds by default.
+
+The key differences to plain `cmake` are : 
+
+  * [`--remote`](#hermetic---remote---host) Builds remotely in an isolated + hermetic environment
+  * [`--host`](#hermetic---remote---host) Disable local containerized build. Builds locally on this host without isolation or hermeticity but with caching.
+  * [`--monitor|-m`](#--monitor-m) Monitors source tree, rebuilding on every changes.
+
+## `cmake-re` : Configure
+Configures and Generate the build system. By default CMake RE relies on Ninja
+```
+cmake-re [<<path-to-source>>]
+  [-S <<path-to-source>]
+  [-B <<path-to-build-dir>>] 
+
+  -DCMAKE_TOOLCHAIN_FILE=<toolchain-file>
+
+  [-D<<var>[:<type>]=<value>>]
+  [-C <<path-to-initial-CMakeCache.txt>>] 
+
+  [-G <generator>] 
+  [--install-prefix <<path-to-install-dir>>]
+
+  [--remote]
+  [--host]
+  [-j|--jobs <cpus>]   
+```
+
+### hermetic, `--remote`, `--host`
+Not specifying any of these options launches a containerized build by default.
+* `--remote`                Builds remotely in an isolated + hermetic environment.
+* `--host`                  Disable local containerized build. Builds locally on this host without isolation or hermeticity but with caching.
+* `-j, --jobs <cpus>`       How many CPU cores have to be dedicated to the build.
+
+### Mandatory
+* `-S <<path-to-source>`    Path to directory with the `CMakeLists.txt` file of the CMake RE project to build.
+* `-B <<path-to-build-dir>>` Path to directory which CMake RE will use as the root of build directory.
+* `-DCMAKE_TOOLCHAIN_FILE=<toolchain-file>` specifies on and for which [environment](./0400-environments.md) to build 
+
+### Optional
+* `-D<<var>[:<type>]=<value>>` Create or update a CMakeCache.txt variable entry.
+* `-C <<path-to-initial-CMakeCache.txt>>` Pre-load a script to populate the cache.
+* `-G <generator>`          Specify a build system generator. Defaults to Ninja ( CMake RE cache-capable fork of ninja )
+* `--install-prefix <<path-to-install-dir>>` Specify the installation directory. Must be an absolute path. `[CMAKE_INSTALL_PREFIX]`
+  
+
+
+### `--monitor, -m`
+* `--monitor, -m`           Monitors source tree, rebuilding on every changes, reconfiguring as needed.
+
+ 
+## `--build <<build-dir>>` : Build
```

Comment:
The order seems odd. Wouldn't `## Build: --build` be more natural? If you change it here, also below.

Reply:
Is this page auto-generated from the CLI?

Reply:
Not exactly in this order for the moment, but this is planned.